components/homme/src/share/prim_driver_base.F90

```diff
@@ -283,13 +291,17 @@ subroutine prim_init1_geometry(elem, par, dom_mt)
     ! ===============================================================
     ! Allocate and initialize the graph (array of GridVertex_t types)
     ! ===============================================================
+#ifndef HOMME_WITHOUT_PIOLIBRARY
```

Comment:
This seems to say that if the user requested an unstructured mesh (e.g., RRM), and PIO isn't available, then it will silently build a structured mesh. Is that true? If so, might it be better to throw an informative error the first time that issue is hit? (Ex: "Mesh files are not supported when BUILD_HOMME_WITHOUT_PIOLIBRARY=TRUE.")

Reply:
So, should I throw this message at config time? Homme does not build with/without meshes, the same exec usually should handle both. This 'without pio' option is not done very nicely, but it serves its purpose and we won't use it regularly.

Reply:
IMO, throw the error at runtime when the netcdf-based mesh logic first starts.